Analysis/src/SignalObjectSelector.cpp

```diff
@@ -38,24 +38,100 @@ SignalObjectSelector::SignalObjectSelector(SignalMode _mode) : mode(_mode)
+{
+    static const std::map<std::pair<Channel, SignalMode>, std::pair<TauIdDiscriminator, DiscriminatorWP>> tauVSmuDiscriminators = {
+        { {Channel::ETau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::HTT_sync}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::HTT}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::HTT}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::HTT}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::TauPOG_default}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::TauPOG_deepTauVsJet}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::TauPOG_deepTauVsJet_full}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::MuTau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Tight} },
+        { {Channel::TauTau, SignalMode::HH_legacy}, {TauIdDiscriminator::againstMuon3, DiscriminatorWP::Loose} },
+        { {Channel::ETau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::VLoose} },
+        { {Channel::MuTau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::VLoose} },
+        { {Channel::TauTau, SignalMode::HH}, {TauIdDiscriminator::byDeepTau2017v2p1VSmu, DiscriminatorWP::VLoose} }
+        // { {Channel::ETau, SignalMode::Skimmer}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::Tight} },
+        // { {Channel::ETau, SignalMode::TauPOG_Skimmer}, {TauIdDiscriminator::byDeepTau2017v2p1VSe, DiscriminatorWP::Tight} },
+    };
+    std::pair<TauIdDiscriminator, DiscriminatorWP> disc_wp;
```

Comment:
it is a map, don't need to loop. use tauVSmuDiscriminators.find. If not found -> throw an exception